### PR TITLE
Csr feat/unit tests

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
Add unit tests for the `AwsIotProvisioning_CreateCertificateFromCsr` API function

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c [hunk 1]
```diff
@@ -216,12 +221,63 @@ static AwsIotProvisioningCreateKeysAndCertificateResponse_t _expectedCreateKeysA
...
+    0x6E,                                                                               /* # text( 14 ) */
+    0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x50, 0x65, 0x6D, /* # "certificatePem" */
+    0x67,                                                                               /* # text(7) */
+    0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67,                                           /* # "abcdefg" */
+    0x6D,                                                                               /* # text(13) */
+    0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x49, 0x64,       /* # "certificateId" */
+    0x66,                                                                               /* # text(6) */
+    0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D,                                                 /* # "hijklm" */
+    0x78, 0x19,                                                                         /*# text(25) */
+    0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x4F, 0x77, 0x6E,
```
Comment (aggarg) [inline libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: nitpick - # alignment.

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c [hunk 2]
```diff
@@ -689,6 +785,39 @@ TEST( Provisioning_Unit_API, Init )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Verifies that the @ref provisioning_function_init function can be
+ * called multiple times though the library resources are created only once.
+ * (If more library resources are created than destroyed, there will be a memory
```
Comment (aggarg) [inline libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Add newline and end with a period.

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c [hunk 3]
```diff
@@ -689,6 +785,39 @@ TEST( Provisioning_Unit_API, Init )
...
+ * @brief Verifies that the library API can gracefully handle calling the CleanUp
+ * and Init functions in reverse order.
+ */
+TEST( Provisioning_Unit_API, CleanupBeforeInit )
+{
+    /* First clean up library to nullify the library init in Test Setup. */
+    AwsIotProvisioning_Cleanup();
+
+    /* Now that library is cleaned up/uninitialized, we will call */
+    /* CleanUp before Init. */
```
Comment (aggarg) [inline libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: ```CleanUp``` --> ```Cleanup```

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c [hunk 4]
```diff
@@ -938,31 +1067,168 @@ TEST( Provisioning_Unit_API, CreateKeysAndCertificateAPIServerResponseAfterTimeo
...
+                                                          &callbackInfo );
+    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_BAD_PARAMETER, status );
+
+    /* Invalid CSR data. */
+    status = AwsIotProvisioning_CreateCertificateFromCsr( _pMqttConnection,
+                                                          IOT_MQTT_QOS_0,
+                                                          NULL,
+                                                          sizeof( _testCsrString ),
+                                                          0,
+                                                          &callbackInfo );
```
Comment (aggarg) [inline libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Should the ```status``` not be verified after this call?

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c [hunk 5]
```diff
@@ -938,31 +1067,168 @@ TEST( Provisioning_Unit_API, CreateKeysAndCertificateAPIServerResponseAfterTimeo
...
+{
+    const uint8_t responseWithMissingEntries[] =
+    {
+        0xA1,                                                             /* # map( 1 ) */
+        0x78, 0x19,                                                       /*# text(25) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x4F, 0x77, 0x6E,
+        0x65, 0x72, 0x73, 0x68, 0x69, 0x70, 0x54, 0x6F, 0x6B, 0x65, 0x6E, /*# "certificateOwnershipToken"
+                                                                           * */
+        0x66,                                                             /*# text(6) */
+        0x54, 0x6F, 0x6B, 0x65, 0x6E, 0x21                                /*# "Token!" */
```
Comment (aggarg) [inline libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: nitpick - ```#``` alignment.

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 4 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Good catch.

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 1 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Fixed

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 2 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Done

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 3 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Done

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 4 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Added

File: libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c (hunk 5 above)
Comment (aggarw13) [reply libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c]: Fixed

**Pull request merged**